net/http/test/CMakeLists.txt

```diff
@@ -4,8 +4,9 @@ target_link_libraries(client_perf PRIVATE photon_shared)
 add_executable(server_perf server_perf.cpp)
 target_link_libraries(server_perf PRIVATE photon_shared)
 
-add_executable(pure_libcurl pure_libcurl.cpp)
-target_link_libraries(pure_libcurl PRIVATE photon_shared)
+if (PHOTON_ENABLE_LIBCURL)
```

Comment:
`header_test` now  used libcurl too, by including ../../curl.h

it should also add macro to make it possible build without PHOTON_ENABLE_LIBCURL